provider/resource.go

```diff
@@ -301,37 +307,38 @@ func arrayValue(vars resource.PropertyMap, prop resource.PropertyKey, envs []str
 // preConfigureCallback returns an error when cloud provider setup is misconfigured
 func preConfigureCallback(vars resource.PropertyMap, c tfshim.ResourceConfig) error {
 
+	// Default environment to Public unless specified.
+	environment := environments.AzurePublic()
 	envName := stringValue(vars, "environment", []string{"ARM_ENVIRONMENT"})
-	if envName == "" {
-		envName = "public"
+	switch cases.Title(language.English).String(envName) {
+	case environments.AzureChinaCloud:
+		environment = environments.AzureChina()
+	case environments.AzureUSGovernmentCloud:
+		environment = environments.AzureUSGovernment()
 	}
 
 	//check for auxiliary tenants
 	auxTenants := arrayValue(vars, "auxiliaryTenantIDs", []string{"ARM_AUXILIARY_TENANT_IDS"})
 
 	// validate the azure config
-	// make a Builder
-	builder := &authentication.Builder{
-		SubscriptionID:       stringValue(vars, "subscriptionId", []string{"ARM_SUBSCRIPTION_ID"}),
-		ClientID:             stringValue(vars, "clientId", []string{"ARM_CLIENT_ID"}),
-		ClientSecret:         stringValue(vars, "clientSecret", []string{"ARM_CLIENT_SECRET"}),
-		TenantID:             stringValue(vars, "tenantId", []string{"ARM_TENANT_ID"}),
-		Environment:          envName,
-		ClientCertPath:       stringValue(vars, "clientCertificatePath", []string{"ARM_CLIENT_CERTIFICATE_PATH"}),
-		ClientCertPassword:   stringValue(vars, "clientCertificatePassword", []string{"ARM_CLIENT_CERTIFICATE_PASSWORD"}),
-		MsiEndpoint:          stringValue(vars, "msiEndpoint", []string{"ARM_MSI_ENDPOINT"}),
-		AuxiliaryTenantIDs:   auxTenants,
-		ClientSecretDocsLink: "https://www.pulumi.com/docs/intro/cloud-providers/azure/setup/#service-principal-authentication",
+	credentials := auth.Credentials{
```

Comment:
Switching to "github.com/hashicorp/go-azure-sdk/sdk/auth" as the authentication builder we previously used is now empty at the latest version.